benchmarks/tests/test_example_benchmarks.py

```diff
@@ -245,22 +173,97 @@ def test_external_cli():
 def test_r_only():
     benchmark = _example_benchmarks.WithoutPythonBenchmark()
     [(result, output)] = benchmark.run()
-    assert_r_only_benchmark(result)
+
+    munged = copy.deepcopy(result)
+    assert munged["tags"] == {
+        "name": "example-R-only",
+        "language": "R",
+        "cpu_count": None,
+    }
+    _asserts.assert_info_and_context(munged, language="R")
     assert _asserts.R_CLI in str(output)
 
 
-def test_r_only_exception():
-    benchmark = _example_benchmarks.BenchmarkExceptionR()
+def test_r_only_nonexistent():
+    benchmark = _example_benchmarks.BenchmarkNonexistentR()
     [(result, output)] = benchmark.run()
-    assert_r_only_benchmark_exception(result)
+
+    munged = copy.deepcopy(result)
+    _asserts.assert_info_and_context(munged, language="R")
+    assert munged["tags"] == {
+        "name": "example-R-only-nonexistent",
+        "cpu_count": None,
+        "language": "R",
+    }
+    command = "run_one(arrowbench:::foo)"
+    assert benchmark._get_r_command() == f"library(arrowbench); {command}"
+    assert "object 'foo' not found" in munged["error"]["error"]
     assert output is None
 
 
-def test_r_only_exception_no_result():
-    benchmark = _example_benchmarks.BenchmarkExceptionNoResultR()
-    [(result, output)] = benchmark.run()
-    assert_r_only_benchmark_exception_no_result(result)
-    assert output is None
+@pytest.mark.parametrize("error_type", ["NULL"])
+@pytest.mark.parametrize("output_type", ["NULL", "'warning'"])
+def test_r_only_placebo(error_type, output_type):
+    benchmark = _example_benchmarks.BenchmarkPlaceboR()
+    [(result, output)] = benchmark.run(case=(error_type, output_type))
+
+    munged = copy.deepcopy(result)
+    _asserts.assert_info_and_context(munged, language="R")
+    assert munged["tags"] == {
+        "name": "example-R-only-exception",
+        "error_type": error_type,
+        "output_type": output_type,
+        "cpu_count": None,
+        "language": "R",
+    }
+    command = (
+        "run_one(arrowbench:::placebo, "
+        f"error_type={error_type}, "
+        f"output_type={output_type})"
+    )
+    assert (
+        benchmark._get_r_command((error_type, output_type))
+        == f"library(arrowbench); {command}"
+    )
+    assert "error" not in munged
+    assert float(munged["stats"]["data"][0]) > 0.0
+    if output_type == "'warning'":
+        assert "A warning" in munged["optional_benchmark_info"]["output"]
+    else:
+        assert "A warning" not in munged["optional_benchmark_info"]["output"]
+    assert command in output
+
+
+@pytest.mark.parametrize("error_type", ["'base'"])
+@pytest.mark.parametrize("output_type", ["NULL", "'warning'"])
+def test_r_only_exception(error_type: str, output_type: str):
```

Comment:
This is a significant refactor of an outdated test which no longer raised errors like before because arrowbench handles them properly. Now this tests that labs/benchmarks handles errors arrowbench catches properly.